tips/TIP-0025/core-rest-api.yaml

```diff
@@ -0,0 +1,3354 @@
+
+    BadRequestResponse:
+      description: Indicates that the request was bad.
+      allOf:
+        - $ref: '#/components/schemas/ErrorResponse'
+      example:
+        error:
+          code: 400
+          message: invalid data provided
+
+    NotFoundResponse:
+      description: Indicates that the data was not found.
+      allOf:
+        - $ref: '#/components/schemas/ErrorResponse'
+      example:
+        error:
+          code: 404
+          message: could not find data
+
+    InternalErrorResponse:
+      description: Indicates that the server encountered an unexpected condition, which prevented it from fulfilling the request by the client.
+      allOf:
+        - $ref: '#/components/schemas/ErrorResponse'
+      example:
+        error:
+          code: 500
+          message: internal server error
+
+    RoutesResponse:
+      description: Contains the available API routes of the node.
+      properties:
+        routes:
+          type: array
+          items:
+            type: string
+      required:
+        - routes
+
+    InfoResponse:
+      description: Returns general information about the node.
+      properties:
+        name:
+          type: string
+          description: The name of the node.
+        version:
+          type: string
+          description: The semantic version of the node.
+        status:
+          description: Status of the node.
+          properties:
+            isHealthy:
+              type: boolean
+              description: Tells whether the node is healthy or not.
+            latestMilestone:
+              type: object
+              description: Information about the latest seen milestone.
+              properties:
+                index:
+                  type: integer
+                  description: The index of the latest seen milestone.
+                timestamp:
+                  type: integer
+                  description: The timestamp of the latest seen milestone.
+                milestoneId:
+                  type: string
+                  description: The Milestone ID of the latest seen milestone. Hex-encoded with 0x prefix.
+              required:
+                - index
+                - timestamp
+                - milestoneId
+            confirmedMilestone:
+              type: object
+              description: Information about the latest confirmed milestone.
+              properties:
+                index:
+                  type: integer
+                  description: The index of the latest confirmed milestone.
+                timestamp:
+                  type: integer
+                  description: The timestamp of the latest confirmed milestone.
+                milestoneId:
+                  type: string
+                  description: The Milestone ID of the latest confirmed milestone. Hex-encoded with 0x prefix.
+              required:
+                - index
+                - timestamp
+                - milestoneId
```

Comment:
```suggestion
```
There is an edge case where this info is not known in the node, so IMO it is better to omit these fields.
This happens for example if there is no milestone issued after bootstrapping from a genesis snapshot file, or after loading a snapshot file.